# EDA
Julian Leonardo Avila Martinez
20212107030

In [11]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

url = "https://raw.githubusercontent.com/lorey/list-of-countries/master/csv/countries.csv"
df = pd.read_csv(url, sep=";")

df.head()

alpha_2 alpha_3      area           capital continent currency_code  \
0      AD     AND     468.0  Andorra la Vella        EU           EUR   
1      AE     ARE   82880.0         Abu Dhabi        AS           AED   
2      AF     AFG  647500.0             Kabul        AS           AFN   
3      AG     ATG     443.0        St. John's       NaN           XCD   
4      AI     AIA     102.0        The Valley       NaN           XCD   

  currency_name eqivalent_fips_code fips  geoname_id          languages  \
0          Euro                 NaN   AN     3041565                 ca   
1        Dirham                 NaN   AE      290557  ar-AE,fa,en,hi,ur   
2       Afghani                 NaN   AF     1149361  fa-AF,ps,uz-AF,tk   
3        Dollar                 NaN   AC     3576396              en-AG   
4        Dollar                 NaN   AV     3573511              en-AI   

                   name         neighbours  numeric   phone  population  \
0               Andorra              ES,FR       20     376       84000   
1  United Arab Emirates              SA,OM      784     971     4975593   
2           Afghanistan  TM,CN,IR,TJ,PK,UZ        4      93    29121286   
3   Antigua and Barbuda                NaN       28  +1-268       86754   
4              Anguilla                NaN      660  +1-264       13254   

  postal_code_format postal_code_regex  tld  
0              AD###  ^(?:AD)*(\d{3})$  .ad  
1                NaN               NaN  .ae  
2                NaN               NaN  .af  
3                NaN               NaN  .ag  
4                NaN               NaN  .ai

In [12]:
df.shape

(252, 19)

In [13]:
df.columns

Index(['alpha_2', 'alpha_3', 'area', 'capital', 'continent', 'currency_code',
       'currency_name', 'eqivalent_fips_code', 'fips', 'geoname_id',
       'languages', 'name', 'neighbours', 'numeric', 'phone', 'population',
       'postal_code_format', 'postal_code_regex', 'tld'],
      dtype='object')

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 252 entries, 0 to 251
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   alpha_2              251 non-null    object 
 1   alpha_3              252 non-null    object 
 2   area                 252 non-null    float64
 3   capital              246 non-null    object 
 4   continent            210 non-null    object 
 5   currency_code        251 non-null    object 
 6   currency_name        251 non-null    object 
 7   eqivalent_fips_code  1 non-null      object 
 8   fips                 249 non-null    object 
 9   geoname_id           252 non-null    int64  
 10  languages            249 non-null    object 
 11  name                 252 non-null    object 
 12  neighbours           165 non-null    object 
 13  numeric              252 non-null    int64  
 14  phone                247 non-null    object 
 15  population           252 non-null    int

The data shows a list of 252 countries with information about certain aspects
such as Name, Capital, Continent, Area, Codes for telephone communication,
shipping, internet extension; Languages spoken on the territory and more.

One can see that the data types seem to be consistent, being quantities the
expected features, one may expect the phone to be a number type, however due
to "+" extension it is treated as an object or string.

There is a big amount of missing data, specially for the postal services.
Some capitals are missing, some continents and neighbours are missing as well
this could be due to being island nation without a clear continent of pertinence
and the lack of land borders with other nations.

Most notably there is one one country that has an equivalent flip code.

In [15]:
df.describe()

area    geoname_id     numeric    population
count  2.520000e+02  2.520000e+02  252.000000  2.520000e+02
mean   5.952879e+05  2.427870e+06  434.309524  2.727679e+07
std    1.904818e+06  1.632093e+06  254.663139  1.164127e+08
min    0.000000e+00  4.951800e+04    0.000000  0.000000e+00
25%    1.098000e+03  1.163774e+06  217.000000  1.879528e+05
50%    6.489450e+04  2.367967e+06  436.000000  4.268583e+06
75%    3.622245e+05  3.478296e+06  652.500000  1.536688e+07
max    1.710000e+07  8.505033e+06  894.000000  1.330044e+09

The numeric info one has is:
- Area - Land are by its territory
- GeoName ID - Number that identifies the location
- Numeric - Numeric code that identifies country for communications
- Population - Amount of inhabitants

Due to the nature of the data, the only ones that could be related are the
area and the population, as countries with larger area are more likely to have
a bigger population.

In [17]:
df_numeric = df[["area", "geoname_id", "numeric", "population"]]
df_numeric

area  geoname_id  numeric  population
0        468.0     3041565       20       84000
1      82880.0      290557      784     4975593
2     647500.0     1149361        4    29121286
3        443.0     3576396       28       86754
4        102.0     3573511      660       13254
..         ...         ...      ...         ...
247  1219912.0      953987      710    49000000
248   752614.0      895949      894    13460305
249   390580.0      878675      716    13061000
250   102350.0     8505033      891    10829175
251      960.0     8505032      530      300000

[252 rows x 4 columns]

In [19]:
corr_var = df_numeric.corr()
print(corr_var)

                area  geoname_id   numeric  population
area        1.000000    0.120399 -0.089264    0.412864
geoname_id  0.120399    1.000000  0.038707   -0.066105
numeric    -0.089264    0.038707  1.000000   -0.041140
population  0.412864   -0.066105 -0.041140    1.000000
